File: solutions/objective-c/leap/1/Leap.m
#import <Foundation/Foundation.h>
#import "Leap.h"
@implementation Leap
- (id)initWithCalendarYear:(NSNumber *)year {
    if (self = [super init]) {
        BOOL divisibleByFour = [year intValue] % 4 == 0;
        BOOL divisibleByHundred = [year intValue] % 100 == 0;
        BOOL divisibleByFourHundred = [year intValue] % 400 == 0;
        if (divisibleByFourHundred || (divisibleByFour && !divisibleByHundred)) {
            self.isLeapYear = YES;
        } else {
            self.isLeapYear = NO;
        }
    }
    return self;
}
@end

File: solutions/cpp/leap/1/leap.cpp
#include "leap.h"
bool leap::is_leap_year(uint16_t year) {
  auto div400 = !(year % 400);
  auto div100 = !(year % 100);
  auto div4 = !(year % 4);
  return div4 && (!div100 || div400);
}
